crates/bifrost/src/bifrost.rs

```diff
@@ -405,24 +405,33 @@ impl BifrostInner {
         Ok(trim_point.unwrap_or(Lsn::INVALID))
     }
 
-    pub async fn trim(&self, log_id: LogId, trim_point: Lsn) -> Result<(), Error> {
+    /// Trim the log to the specified LSN trim point (inclusive). Returns the new trim point LSN if
+    /// the log was actually trimmed by this call, or `None` otherwise.
+    pub async fn trim(&self, log_id: LogId, trim_point: Lsn) -> Result<Option<Lsn>, Error> {
```

Comment:
I am wondering whether changing this contract t always return the lowest available lsn is a bit nicer. In the current form if I send trim(0, 5) twice, my first request will receive a `Some(5)` as response and the second `None`. Differently asked, why is it important to be able to distinguish whether a `trim` call has actually performed the trim or not?